src/views/virtualmachines/details/components/disk/hooks/useFilesystemListGuestOS.ts

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+
+import VirtualMachineInstanceModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineInstanceModel';
+import {
+  V1VirtualMachineInstance,
+  V1VirtualMachineInstanceGuestAgentInfo,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { consoleFetch } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseFileSystemTableGuestOS = (
+  vmi?: V1VirtualMachineInstance,
+) => [V1VirtualMachineInstanceGuestAgentInfo, boolean, any];
+
+const useFilesystemTableGuestOS: UseFileSystemTableGuestOS = (vmi) => {
+  const [loaded, setLoaded] = React.useState(false);
+  const [data, setData] = React.useState<V1VirtualMachineInstanceGuestAgentInfo>({});
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    const guestOS = vmi?.status?.guestOSInfo?.id;
+
+    setError(null);
+    if (guestOS) {
+      (async () => {
+        const response = await consoleFetch(
+          `api/kubernetes/apis/subresources.${VirtualMachineInstanceModel.apiGroup}/${VirtualMachineInstanceModel.apiVersion}/namespaces/${vmi?.metadata?.namespace}/${VirtualMachineInstanceModel.plural}/${vmi?.metadata?.name}/guestosinfo`,
+        );
+        const jsonData = await response.json();
+        setData(jsonData);
+        setLoaded(true);
+      })().catch((err) => {
+        setError(err);
+        setLoaded(true);
+      });
+    }
+    (!vmi || (!guestOS && vmi?.metadata)) && setLoaded(true);
```

Comment:
vmi or vm ? :) 

also, windows machines dont have guestOS so it safer to check !guestOS && vmi?.metadata and not !vmi || (!guestOS && vmi?.metadata)

Reply:
vmi is the one consisting the guestOSInfo if there is any, so it should be vmi :) 
this check is provided in order to get the right emptyMsg for the filesystem table on the vm view by the FilesystemList component: https://github.com/kubevirt-ui/kubevirt-plugin/pull/85/files#diff-2450440d82a58d237fd0684fe42248b9a1dba40f7fc5a82413d3a8aac3b0169a

but I do need to add the check you suggested to that component so will change :)